cmd/dependabot/internal/cmd/update.go

```diff
@@ -156,6 +85,86 @@ var updateCmd = &cobra.Command{
 	},
 }
 
+func extractInput(cmd *cobra.Command) (*model.Input, []byte, error) {
+	var input *model.Input
+	var inputRaw []byte
+
+	if file != "" {
+		if len(cmd.Flags().Args()) > 0 {
+			return nil, nil, errors.New("cannot use file and arguments together")
+		}
+		var err error
+		input, inputRaw, err = readInputFile(file)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	if len(cmd.Flags().Args()) > 0 {
+		if len(cmd.Flags().Args()) != 2 {
+			return nil, nil, errors.New("requires a package manager and repo argument")
+		}
+
+		packageManager = cmd.Flags().Args()[0]
+		if packageManager == "" {
+			return nil, nil, errors.New("requires a package manager argument")
+		}
+
+		repo = cmd.Flags().Args()[1]
+		if repo == "" {
+			return nil, nil, errors.New("requires a repo argument")
+		}
+
+		input = &model.Input{
+			Job: model.Job{
+				PackageManager: packageManager,
+				AllowedUpdates: []model.Allowed{{
+					UpdateType: "all",
+				}},
+				DependencyGroups:           nil,
+				Dependencies:               nil,
+				ExistingPullRequests:       [][]model.ExistingPR{},
+				IgnoreConditions:           []model.Condition{},
+				LockfileOnly:               false,
+				RequirementsUpdateStrategy: nil,
+				SecurityAdvisories:         []model.Advisory{},
+				SecurityUpdatesOnly:        false,
+				Source: model.Source{
+					Provider:    provider,
+					Repo:        repo,
+					Directory:   directory,
+					Branch:      nil,
+					Hostname:    nil,
+					APIEndpoint: nil,
+				},
+				UpdateSubdependencies: false,
+				UpdatingAPullRequest:  false,
+			},
+		}
+	}
+
+	if inputServerPort != 0 {
+		input = server.Input(inputServerPort)
+	}
+
+	if doesStdinHaveData() {
+		in := &bytes.Buffer{}
+		_, err := io.Copy(in, os.Stdin)
+		if err != nil {
+			return nil, nil, err
+		}
+		data := in.Bytes()
+		input = &model.Input{}
+		if err = json.Unmarshal(data, &input); err != nil {
+			if err = yaml.Unmarshal(data, &input); err != nil {
+				return nil, nil, fmt.Errorf("failed to decode input file: %w", err)
+			}
+		}
+	}
+
+	return input, inputRaw, nil
```

Comment:
TBH, if I'm spelunking in the code, if I see this and I have to figure out if the source of this input is stdin, cmd flags, network server, or a file, then i'm really unclear unless i read the whole function above, which has a lot of code that doesn't matter if i'm just trying to figure out what "wins" as the source of truth...